Engine/game/savegame.cpp

```diff
@@ -0,0 +1,596 @@
+//=============================================================================
+//
+// Adventure Game Studio (AGS)
+//
+// Copyright (C) 1999-2011 Chris Jones and 2011-20xx others
+// The full list of copyright holders can be found in the Copyright.txt
+// file, which is part of this source code distribution.
+//
+// The AGS source code is provided under the Artistic License 2.0.
+// A copy of this license can be found in the file License.txt and at
+// http://www.opensource.org/licenses/artistic-license-2.0.php
+//
+//=============================================================================
+
+#include "ac/character.h"
+#include "ac/common.h"
+#include "ac/draw.h"
+#include "ac/dynamicsprite.h"
+#include "ac/event.h"
+#include "ac/game.h"
+#include "ac/gamesetupstruct.h"
+#include "ac/gamestate.h"
+#include "ac/gamesetup.h"
+#include "ac/global_audio.h"
+#include "ac/global_character.h"
+#include "ac/gui.h"
+#include "ac/mouse.h"
+#include "ac/overlay.h"
+#include "ac/region.h"
+#include "ac/richgamemedia.h"
+#include "ac/room.h"
+#include "ac/roomstatus.h"
+#include "ac/spritecache.h"
+#include "ac/system.h"
+#include "debug/out.h"
+#include "device/mousew32.h"
+#include "gfx/bitmap.h"
+#include "gfx/ddb.h"
+#include "gfx/graphicsdriver.h"
+#include "game/savegame.h"
+#include "main/main.h"
+#include "main/version.h"
+#include "media/audio/audio.h"
+#include "media/audio/soundclip.h"
+#include "platform/base/agsplatformdriver.h"
+#include "plugin/agsplugin.h"
+#include "script/script.h"
+#include "script/cc_error.h"
+#include "util/alignedstream.h"
+#include "util/file.h"
+#include "util/stream.h"
+#include "util/string_utils.h"
+
+
+extern GameSetupStruct game;
+extern Bitmap **guibg;
+extern AGS::Engine::IDriverDependantBitmap **guibgbmp;
+extern AGS::Engine::IGraphicsDriver *gfxDriver;
+extern Bitmap *dynamicallyCreatedSurfaces[MAX_DYNAMIC_SURFACES];
+extern Bitmap *raw_saved_screen;
+extern RoomStatus troom;
+extern RoomStatus *croom;
```

Comment:
Actually, could these globals be passed in as references to the save functions?  It would make it more explicit that we're changing global state. (maybe a different pull request)

It's messy but I like that we're being explicit at the top which globals we're using.